velox/type/Variant.cpp

```diff
@@ -236,11 +250,18 @@ std::string variant::toJson(const TypePtr& type) const {
       std::string b{};
       b += "[";
       bool first = true;
+      uint32_t idx = 0;
```

Comment:
Array can have only a single elementType as its child.
We should enforce that here and remove the `for` loop below.

Reply:
@majetideepak - I did the change to enforce it takes only single element type into consideration for ARRAY, but the for loop remains. Could you please confirm if this is sufficient here?
```
    case TypeKind::ARRAY: {
      auto& array = value<TypeKind::ARRAY>();
      std::string b{};
      b += "[";
      bool first = true;
      uint32_t idx = 0;
      auto arrayElementType = type->childAt(0);
      for (uint32_t idx = 0; idx < array.size(); ++idx)
      {
        if(!first) {
          b += ",";
        }
        b += array.at(idx).toJson(arrayElementType);
        first = false;
      }
      b += "]";
      return b;
    }
```

I also did the testing and works fine
```
  // Array variant tests
  auto arrayTypePtr = ARRAY(DECIMAL(20, 3));
  EXPECT_EQ(
      "[123456.789,634565.464,234545.278]",
      variant::array({variant((int128_t)123456789),
                      variant((int128_t)634565464),
                      variant((int128_t)234545278)})
          .toJson(arrayTypePtr));
  EXPECT_THROW( 
      variant::array({variant((int128_t)123456789),
                      variant((std::string) "Array should not support multiple element types.")})
          .toJson(arrayTypePtr),
      VeloxUserError);
```

Reply:
Why do we need the for loop for Array type? Can you explain?

Reply:
For example, suppose we have 3 elements of a type kind, then the toJson() function should run for each element. This is what my understanding is, isn't it?
This is example I tested.
```
auto arrayTypePtr = ARRAY(DECIMAL(20, 3));
  EXPECT_EQ(
      "[123456.789,634565.464,234545.278]",
      variant::array({variant((int128_t)123456789),
                      variant((int128_t)634565464),
                      variant((int128_t)234545278)})
          .toJson(arrayTypePtr));
```

Reply:
@Krishna-Prasad-P-V you are right. Thanks for the example.